VirtoCommerce.MarketingModule.Web/Controllers/Api/MarketingModulePromotionController.cs

```diff
@@ -201,7 +207,13 @@ public IHttpActionResult DeletePromotions([FromUri] string[] ids)
         public IHttpActionResult SearchCoupons(CouponSearchCriteria criteria)
         {
             var searchResult = _couponService.SearchCoupons(criteria);
-
+            // actualize coupon totalUsage field with out caching 
+            using (var repository = _repositoryFactory())
+            {
+                var ids = searchResult.Results.Select(x => x.Id).ToArray();
+                var couponEntities = repository.GetCouponsByIds(ids);
+                searchResult.Results.ForEach(coupon => coupon.TotalUsesCount = couponEntities.First(c => c.Id == coupon.Id).TotalUsesCount);
```

Comment:
Because Coupon is  inherited from Entity class and has an override GetHashCode and Equals methods, you can replace `couponEntities.First(c => c.Id == coupon.Id)` with `couponEntities.First(c => c == coupon or c.equals(coupon))

Reply:
Coupon class doesn't override Equals method.

Reply:
https://github.com/VirtoCommerce/vc-module-core/blob/master/VirtoCommerce.Domain/Marketing/Model/Promotions/Coupon.cs#L5
Coupon <- AuditableEntity <- Entity
https://github.com/VirtoCommerce/vc-platform/blob/master/VirtoCommerce.Platform.Core/Common/Entity.cs#L24

Reply:
Ok. but nevertheless RealObjectType is different. 
https://github.com/VirtoCommerce/vc-platform/blob/e77d5fe8468d20bdcc4d5107f53593ddfec77441/VirtoCommerce.Platform.Core/Common/Entity.cs#L33

Reply:
Sorry, I don't understand what you meant.